app/misc/graphing.ts

```diff
@@ -385,11 +415,43 @@ function makeBasicNode(c, column: number) {
             });
 
             bsEdges.add({
-                from: id + numOfCommits * (i + 1),
+                from: bsnodeId,
                 to: id
             });
 
-            branchIds[id] = id + numOfCommits * (i + 1);
+            branchIds[id] = bsnodeId;
+        }
+    }
+
+    // Initializing viewable tags in highest zoom graph level
+    if (c.toString() in tags) {
+        for (let i = 0; i < tags[c.toString()].length; i++) {
```

Comment:
I'm sure the compiler (or whatever this thing uses) optimizes this code; however, you can cache 'tags[c.toString()].length' as opposed to recalculating it every iteration. I don't know how expensive a.) c.toString() is, b.) the lookup should be constant, and c.) .length could be O(n).

Reply:
That's a good thought. They do the same for the branches, however I can add a variable at the beginning of each if statement to calculate this for branches, and tags.